dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/expression/DefaultExpressionService.java

```diff
@@ -674,20 +708,59 @@ private Object visit( String expression, DataType dataType, CommonExpressionVisi
      * substituted for constants and orgUnitCounts.
      *
      * @param expression the expresion to regenerate.
-     * @param constantMap map of constants to use for calculation.
-     * @param orgUnitCountMap the map of organisation unit group member counts.
+     * @param constants map of constants to use for calculation.
+     * @param orgUnitGroups map of organisation unit groups.
      * @return the regenerated expression string.
      */
     private String regenerateIndicatorExpression( String expression,
-        Map<String, Constant> constantMap, Map<String, Integer> orgUnitCountMap )
+        Map<String, Constant> constants, Map<String, OrganisationUnitGroup> orgUnitGroups )
     {
-        CommonExpressionVisitor visitor = newVisitor( INDICATOR_EXPRESSION, ITEM_REGENERATE,
-            DEFAULT_SAMPLE_PERIODS, constantMap, NEVER_SKIP );
+        if ( expression == null || expression.isEmpty() )
+        {
+            return null;
+        }
 
-        visitor.setOrgUnitCountMap( orgUnitCountMap );
-        visitor.setExpressionLiteral( new RegenerateLiteral() );
+        // ---------------------------------------------------------------------
+        // Constants
+        // ---------------------------------------------------------------------
+
+        StringBuffer sb = new StringBuffer();
+        Matcher matcher = CONSTANT_PATTERN.matcher( expression );
+
+        while ( matcher.find() )
+        {
+            String co = matcher.group( GROUP_ID );
+
+            Constant constant = constants.get( co );
+
+            String replacement = constant != null ? String.valueOf( constant.getValue() ) : NULL_REPLACEMENT;
+
+            matcher.appendReplacement( sb, Matcher.quoteReplacement( replacement ) );
+        }
+
+        expression = TextUtils.appendTail( matcher, sb );
+
+        // ---------------------------------------------------------------------
+        // Org unit groups
+        // ---------------------------------------------------------------------
+
+        sb = new StringBuffer();
+        matcher = OU_GROUP_PATTERN.matcher( expression );
+
+        while ( matcher.find() )
+        {
+            String oug = matcher.group( GROUP_ID );
+
+            OrganisationUnitGroup group = orgUnitGroups.get( oug );
+
+            String replacement = group != null ? String.valueOf( group.getMembers().size() ) : NULL_REPLACEMENT;
+
+            matcher.appendReplacement( sb, replacement );
+
+            // TODO sub tree
```

Comment:
Should the TODO be there? If yes, could you make a jia issue for it and reference it here? Then we can track it from the code next time we stumble over it :)

Reply:
I copied the TODO it from the 2.33 pre-ANTLR code that I was restoring to do this job. I have no idea what it means. I debated whether to remove it or not, and was on the fence about it. I thought it might be better to leave it in case it's meaningful to someone else some day. Alternately we could remove it. I'm fine either way. I'm not aware of any deficiency in the code that this could refer to.